src/partition/repart.c

```diff
@@ -6656,20 +6767,10 @@ static int parse_argv(int argc, char *argv[]) {
                 }
 
                 case ARG_PRIVATE_KEY: {
-                        _cleanup_(erase_and_freep) char *k = NULL;
-                        size_t n = 0;
-
-                        r = read_full_file_full(
-                                        AT_FDCWD, optarg, UINT64_MAX, SIZE_MAX,
-                                        READ_FULL_FILE_SECURE|READ_FULL_FILE_WARN_WORLD_READABLE|READ_FULL_FILE_CONNECT_SOCKET,
-                                        NULL,
-                                        &k, &n);
-                        if (r < 0)
-                                return log_error_errno(r, "Failed to read key file '%s': %m", optarg);
-
-                        EVP_PKEY_free(arg_private_key);
-                        arg_private_key = NULL;
-                        r = parse_private_key(k, n, &arg_private_key);
+                        if (is_path(optarg))
```

Comment:
hmm, this will break a bunch of uses where people just specified `--private-key=foobar.pem`, because that will not return true in `is_path()`.

I'd propose just storing the passed optarg here in a local variable, and then at the end of the getopt() loop check if the engine stuff is set, and if not load a PEM file, and if it is store it as a literal string.